compiler/src/iree/compiler/Codegen/Common/GPU/VectorReductionToGPU.cpp

```diff
@@ -63,15 +64,24 @@ static bool isUniformLoad(Operation *op) {
   auto loadOp = dyn_cast<memref::LoadOp>(op);
   if (!loadOp)
     return false;
-  auto space = loadOp.getMemRefType().getMemorySpace();
-  auto attr = llvm::dyn_cast_if_present<DescriptorTypeAttr>(space);
-  if (!attr)
+  if (!hasGlobalMemoryAddressSpace(loadOp.getMemRefType()))
     return false;
-
-  if (attr.getValue() == DescriptorType::UniformBuffer)
-    return true;
+  auto space = loadOp.getMemRefType().getMemorySpace();
+  auto descTypeAttr = llvm::dyn_cast_if_present<DescriptorTypeAttr>(space);
 
   auto subspan = loadOp.getMemRef().getDefiningOp<InterfaceBindingSubspanOp>();
+  if (auto fatBufferCast =
```

Comment:
Maybe this can be made a bit more cleaner? IIUC if `subspan` is empty, `descTypeAttr` is null anyway. So do this

```
auto descTypeAttr = llvm::dyn_cast_if_present<DescriptorTypeAttr>(space);
if (descTypeAttr && descTypeAttr.getValue() == DescriptorType::UniformBuffer) {
  return true;
}

auto subspan = loadOp.getMemRef().getDefiningOp<InterfaceBindingSubspanOp>()
if (!subspan) {
  if (auto fatBufferCast = ) {
    subspan = fatBufferCast.getSource().getDefiningOp<InterfaceBindingSubspanOp>();
  }
}

if (!subspan) {
  return false;
}
descrTypeAttr = dyn_cast_if_present<...>(subspan.getResult().getType().getMemorySpace())

if (descrTypeAttr && descrTypeAttr.getValue() == DescriptorType::UniformBuffer){
  return true;
}


```